skbio/metadata/_interval.py

```diff
@@ -0,0 +1,756 @@
+        for loc in self.locations:
+            start, end = loc
+            self._interval_metadata._interval_tree.add(start, end, self)
+        self._interval_metadata._intervals.append(self)
+
+    @experimental(as_of='0.5.0-dev')
+    def __eq__(self, other):
+        '''Test if this ``Interval`` object is equal to another.
+
+        The equality is performed by checking if the ``metadata``,
+        ``location`` and ``boundaries`` are equal. Since the ``locations``
+        and the ``boundaries`` are sorted, the permutations of them during
+        the ``Interval`` construction or assignment won't matter.
+
+        Parameters
+        ----------
+        other : Interval
+            Interval to test for equality against.
+
+        Returns
+        -------
+        bool
+            Indicates if the two objects are equal.
+        '''
+        return ((self.metadata == other.metadata) and
+                (self.locations == other.locations) and
+                (self.boundaries == other.boundaries))
+
+    @experimental(as_of='0.5.0-dev')
+    def __ne__(self, other):
+        '''Test if this ``Interval`` object is not equal to another.
+
+        Parameters
+        ----------
+        other : Interval
+            Interval to test for inequality against.
+
+        Returns
+        -------
+        bool
+            Indicates if the two objects are not equal.
+        '''
+        return not self.__eq__(other)
+
+    @experimental(as_of='0.5.0-dev')
+    def __repr__(self):
+        '''Return a string representation of this ``Interval`` object.
+
+        Returns
+        -------
+        str
+            String representation of this ``Interval`` object.
+        '''
+        if self.dropped:
+            s = ('{}(dropped=True, locations={!r}, '
+                 'boundaries={!r}, metadata={!r})')
+            return s.format(self.__class__.__name__,
+                            self.locations, self.boundaries, self.metadata)
+        else:
+            s = ('{}(interval_metadata=<{!r}>, locations={!r}, '
+                 'boundaries={!r}, metadata={!r})')
+            return s.format(self.__class__.__name__,
+                            id(self._interval_metadata),
+                            self.locations, self.boundaries, self.metadata)
+
+    @experimental(as_of='0.5.0-dev')
+    def drop(self):
+        '''Drop this ``Interval`` object from the interval metadata it links to.
+
+        If the ``Interval`` object is dropped, you can still get values of
+        ``locations``, ``boundaries``, and ``metadata`` attributes, but you
+        can not change their values with the setters.
+
+        See Also
+        --------
+        skbio.metadata.IntervalMetadata.drop
+        '''
+        if not self.dropped:
+            self._interval_metadata.drop([self])
+
+    def _locations_boundaries_setter(self, locations=None, boundaries=None):
+        if self.dropped:
+            raise RuntimeError('Cannot change the dropped '
```

Comment:
```
"Cannot change `locations` or `boundaries` on a dropped Interval object."
```

This will make it clear exactly what things can't be changed (the current message implies that `metadata` cannot be changed).